### PR TITLE
add base class of LearningRateEpochDecay, and API: MultiStepDecay, and API: StepDecay

### PR DESCRIPTION
**Pull request by zhwesky2010**
### PR types

New features
### PR changes

APIs
### Describe

New API: 
`paddle.fluid.dygraph.MultiStepDecay` 
`paddle.fluid.dygraph.StepDecay` 

Two new learing rate strategy, support in dygraph mode.
At the same time, the base class of learning rate strategy for EPOCH updated was added.

### CODE REVIEW

File: python/paddle/fluid/layers/learning_rate_scheduler.py [hunk 1]
```diff
@@ -589,3 +589,92 @@ def linear_lr_warmup(learning_rate, warmup_steps, start_lr, end_lr):
...
+        if global_step < 30:
+            learning_rate = 0.5
+        elif global_step < 50:
+            learning_rate = 0.05
+        else:
+            learning_rate = 0.005
+
+    Parameters:
+        learning_rate (Variable|float|int): The initial learning rate. It can be set to python float or int number. If 
+            the type is Variable, it should be 1-D Tensor with shape [1], and the data type should be 'float32' or 'float64'.
```
Comment (lanxianghit) [inline python/paddle/fluid/layers/learning_rate_scheduler.py]: variable也应该支持int类型吧？

File: python/paddle/fluid/layers/learning_rate_scheduler.py [hunk 2]
```diff
@@ -589,3 +589,92 @@ def linear_lr_warmup(learning_rate, warmup_steps, start_lr, end_lr):
...
+        milestones (tuple|list): List or tuple of each boundaries. Must be increasing.
```
Comment (lanxianghit) [inline python/paddle/fluid/layers/learning_rate_scheduler.py]: 是开区间还是闭区间最好说清楚

File: python/paddle/fluid/layers/learning_rate_scheduler.py [hunk 3]
```diff
@@ -589,3 +589,92 @@ def linear_lr_warmup(learning_rate, warmup_steps, start_lr, end_lr):
...
+        decay_rate (float, optional): The Ratio that the learning rate will be reduced. ``new_lr = origin_lr * decay_rate`` . 
+            It should be less than 1.0. Default: 0.1.
```
Comment (lanxianghit) [inline python/paddle/fluid/layers/learning_rate_scheduler.py]: 对参数的各种条件限制好像没有看到在代码端做检查？比如milestone要求“Must be increasing.”，decay_rate要求“It should be less than 1.0.”

File: python/paddle/fluid/layers/learning_rate_scheduler.py [hunk 4]
```diff
@@ -589,3 +589,92 @@ def linear_lr_warmup(learning_rate, warmup_steps, start_lr, end_lr):
...
+        milestones = [3, 5]
+        lr = fluid.layers.multi_step_decay(0.5, milestones, decay_rate=0.1)
+        adam = fluid.optimizer.Adam(learning_rate = lr)
+
+        place = fluid.CPUPlace()
+        exe = fluid.Executor(place)
+        exe.run(fluid.default_startup_program())
+        for step in range(10):
+            out, = exe.run(fluid.default_main_program(), fetch_list=[lr.name])
+            print("step:{}, current lr is {}" .format(step, out))
```
Comment (lanxianghit) [inline python/paddle/fluid/layers/learning_rate_scheduler.py]: 最好展示一下输出的信息

File: python/paddle/fluid/layers/learning_rate_scheduler.py (hunk 1 above)
Comment (zhwesky2010) [reply python/paddle/fluid/layers/learning_rate_scheduler.py]: learning_rate会保存为一个Variable在optimizer中，如果是int类型的Variable，就只能是0和1了。会维持一开始的dtype，不会去强转

File: python/paddle/fluid/layers/learning_rate_scheduler.py (hunk 2 above)
Comment (zhwesky2010) [reply python/paddle/fluid/layers/learning_rate_scheduler.py]: [30, 50]和(30, 50)效果是一样的

File: python/paddle/fluid/layers/learning_rate_scheduler.py (hunk 3 above)
Comment (zhwesky2010) [reply python/paddle/fluid/layers/learning_rate_scheduler.py]: Done

File: python/paddle/fluid/layers/learning_rate_scheduler.py (hunk 4 above)
Comment (zhwesky2010) [reply python/paddle/fluid/layers/learning_rate_scheduler.py]: Done

File: python/paddle/fluid/layers/learning_rate_scheduler.py (hunk 2 above)
Comment (zhwesky2010) [reply python/paddle/fluid/layers/learning_rate_scheduler.py]: list和tuple是一样的，都是在每个元素的位置发生衰减。第30步、第50步

**Review by lanxianghit — DISMISSED** [review]

**Review by phlrain — APPROVED** [review]

**Review by lanxianghit — APPROVED** [review]

**Review by saxon-zh — APPROVED** [review]
LGTM
